File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        int n = nums.length;
        List<List<Integer>> res = new ArrayList<>();
        for(int i = 0;i < (1 << n);i++){
            List<Integer> sub = new ArrayList<>();
            for(int j = 0;j < n;j++){
                if((i & (1 << j)) != 0){
                    sub.add(nums[j]);
                }
            }
            res.add(sub);
        }
        return res;
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        n = len(nums)
        res = []
        
        for i in range(1 << n):  # Iterate through 2^n subsets
            sub = []
            for j in range(n):
                if (i & (1 << j)) != 0:  # Check if j-th bit is set
                    sub.append(nums[j])
            res.append(sub)
        
        return res